internal/consensus/state.go

```diff
@@ -594,16 +594,22 @@ func (cs *State) scheduleTimeout(duration time.Duration, height int64, round int
 }
 
 // send a msg into the receiveRoutine regarding our own proposal, block part, or vote
-func (cs *State) sendInternalMessage(mi msgInfo) {
+func (cs *State) sendInternalMessage(ctx context.Context, mi msgInfo) {
 	select {
+	case <-ctx.Done():
 	case cs.internalMsgQueue <- mi:
 	default:
 		// NOTE: using the go-routine means our votes can
 		// be processed out of order.
 		// TODO: use CList here for strict determinism and
 		// attempt push to internalMsgQueue in receiveRoutine
 		cs.Logger.Debug("internal msg queue is full; using a go-routine")
```

Comment:
Not related to your change, but this log makes me sad.

"I know I'm only allowed to eat fifty pieces of candy, but I've already eaten fifty pieces so I'll just take another one out behind the woodshed and eat it there while nobody is looking."